src/queue.js

```diff
@@ -219,4 +219,6 @@ class Queue extends EventEmitter {
   }
 }
 
+Queue.Job = null; // Make TypeScript compiler happy. Needed because Job is an abstract class.
```

Comment:
I understand. Unfortunately I can't put this in to index.d.ts since that file is only allowed to contain declarations, not executable statements.

And I'm not sure this is an entirely TypeScript-specific problem, in that I suspect that the ES6 class keyword works in a similar manner: when you are inheriting from a class, the thing that you are inheriting from has to be a value.

Again, this might be my inexperience with TypeScript; I created a stackoverflow question specifically about this issue and have gotten no responses.

Part of the problem is that I'm doing something weird here - normally, when you are writing a type definitions file for a javascript library, you declare interfaces, not classes. Interfaces are purely compile-time entities that have no runtime existence, so it doesn't matter if they exist as values or not.

However, in the specific case of Job, declaring it as an interface creates a problem. The way Job is defined is that it is a mix of library properties and application properties. The way to model that is using inheritance - either as a class or as an interface.

The way interfaces work is that when you declare a class as implementing an interface, you are basically telling the compiler that your class conforms to that interface. That means that your concrete class has to re-declare all of the properties of the interface. Given the large number of properties defined in Job, this seems tedious and error-prone.

The alternative is to say that Job is an abstract class rather than an interface. That way, when you inherit from it, you get all the standard properties defined automatically. My rationale for doing this is that what rethinkdb-job-queue is providing is in fact an actual implementation and not just a type.

However, rethinkdb-job-queue library doesn't export Job as an entity, because it doesn't want you creating your own Job objects. It's more like a factory function. But unlike most factory functions, it lets you add arbitrary properties to it, so that the resulting object is a combination of a user type and a library type. Figuring out how to model this is proving to be a challenge.